src/oic/oic/__init__.py

```diff
@@ -628,9 +628,7 @@ def construct_EndSessionRequest(
         if request_args is None:
             request_args = {}
 
-        if "state" in kwargs:
-            request_args["state"] = kwargs["state"]
-        elif "state" in request_args:
+        if "state" in request_args and "state" not in kwargs:
```

Comment:
This is also achievable by using `setdefault("state", kwargs.get("state"))` and you can go without the condition completely.

Althought that would set `state  = None` if it would be missing in both.

But I am not sure if that is readable enough, so leaving that up to you.

Reply:
I'd go for what I have proposed since I think it's more readable.

Reply:
Fine with me.